### PR TITLE
Fix IMRPhenomXAS h22 mode phase convention (issue #9)

### LINKED ISSUE #9: IMRPhenomXAS $\pi$ phase offset
The phase of $h_{22}$ appears to show an offset of $\pi$ between LAL and ripple. This doesn't seem to appear in the plus and cross polarisations though. Does LAL add something further downstream that I'm missing?

This is the $h_{22}$ phase offset for a few runs, I still need to look into the deviations from $\pi$:
Image

Here is the $h_+$ strain. LAL and ripple seem to agree here. Also the phase looks good here (second plot). 
Image

Image

### PR DESCRIPTION
**Pull request by thomasckng**
gen_IMRPhenomXAS now returns the correct h22 mode matching LAL's mode-decomposition convention. Previously, phifRef included a spurious +PI offset that made h0 = -h22_LAL; the polarisation formula compensated with hp = h0·antenna, which gave correct h+/h× but wrong h22.

After the fix:
- gen_IMRPhenomXAS → h22 = h22_LAL (zero phase offset vs XHM)
- gen_IMRPhenomXAS_hphc → hp = -h22·(1+cos²ι)/2 (LAL sign convention)
- h+, h× unchanged; 1000-sample overlap test still passes

Closes #9.

## Release Notes

* **Bug Fixes**
  * Corrected phase conventions and polarisation signs in waveform generation to align with standard conventions.

* **Tests**
  * Added phase convention validation tests to verify waveform accuracy against reference implementations across supported models.

[Review Change Stack](https://app.coderabbit.ai/change-stack/GW-JAX-Team/ripple/pull/122?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by thomasckng** [comment]
@coderabbitai pause

**Review by robkamcha — CHANGES_REQUESTED** [review]
Apologies in advance for my pedantry, I just want to make sure we don't mix up LAL conventions:
This PI-offset in the $h_{22}$ phase was originally added to account for the fact that LAL only adds in the $Y_2^2$ factor (which contains a $\pi$-phase offset) after the $h_{22}$-mode is generated. Perhaps for clarity we can add in a comment detailing that the $-1$-factor comes from $e^{i\pi}$ in $Y_2^2$?
Technically the LAL convention looks like this, where `hc` receives a factor of $-1$:
```c
...
for(j = 0; j < (*hptilde)->data->length; j++)
              {
                (*hctilde)->data->data[j] = -I*cfac * (*hptilde)->data->data[j] * Ylmfactor;
                (*hptilde)->data->data[j] *= [pfac](https://lscsoft.docs.ligo.org/lalsuite/7.26/lalsimulation/_p_n_coefficients_8c.html#abfcca6ca27eec07ffb5bcc9b579dbd40) * Ylmfactor;
              }
...
```

I would suggest a comment like this in `IMRPhenomXAS.py` when calculating `hp` and `hc`:
```python
    # -1 prefactor in hp and hc comes from e^(i PI) in Y22
    hp = -h22 * (1 / 2 * (1 + jnp.cos(iota) ** 2))
    hc = 1j * h22 * jnp.cos(iota)
```

Other than that, this looks a lot clearer than the original version, cheers!

**Comment by thomasckng** [comment]
@robkamcha Just added the comment. Is this what you are thinking?

**Comment by robkamcha** [comment]
@thomasckng looks perfect!
